### DISCUSSION TITLE
condition variable

### DISCUSSION THREAD

**Discussion by kamurj** [AutomaApp/automa#1438] [Q&A]
Hello, inside condition builder when I try to test variable value like this it fails. Any ideas what I am doing wrong? Thanks.

![image](https://github.com/AutomaApp/automa/assets/8271715/6869b77a-cf54-4413-afa1-84e100175b04)

Should I instead format it like this {{variables@$$zoutline}}  ?

**Comment by Delusionnnn** [comment]
What is your variable name? Is it $$zoutline (aka a global variable) or zoutline.

Also what do you mean by failed, do you mean its not recognizing when the variable doesnt equal zzz and instead executing the fallback?

Try not equals (case insenstive) and not contains and see if that solves it as well.

**Comment by kamurj** [comment]
The variable name is zoutline. Case is not an issue as it will never use upper case.

So, should I be using {{variables@$$zoutline}} or {{variables,zoutline}} ?

Thanks.

**Comment by Delusionnnn** [comment]
You should be using {{variables.zoutline}} then.

 Is the log saying the value of the variable before it runs through the condition block? It could be not assigning the variable and therefore leaving it blank which of course will make it not work.

**Comment by kamurj** [comment]
Thank you for clarifying. 

The workflow starts with a loop, then I use 'Get text' to try and extract content outline and store it in zoutline variable. At the very end of each loop the variable zoutline is set to zzz. So if in the workflow it is set to zzz, then I assume outline failed and plan b has to be executed.

When the workflow runs I often don't the correct variable value but I know it is there because the workflow executes as expected but fails when evaluating the value inside condition block.

**Reply by Delusionnnn** [reply]
Ya you need to go to the logs and see what information it gives you near the end otherwise it'll be just guessing
